plugins/module_utils/network/sonic/config/l3_interfaces/l3_interfaces.py

```diff
@@ -183,9 +183,10 @@ def _state_replaced(self, want, have):
         """
         ret_requests = list()
         commands = list()
-        new_want = self.update_object(want)
-        new_have = self.remove_default_entries(have, False)
-        get_replace_interfaces_list = self.get_interface_object_for_replaced(new_have, want)
+        new_have = remove_empties_from_list(have)
+        new_have = self.remove_default_entries(new_have)
+        new_want = self.remove_default_entries(want)
+        get_replace_interfaces_list = self.get_interface_object_for_replaced(new_have, new_want)
```

Comment:
If 'want' had only default values, it would be deleted in 'new_want' and will not be included in the list of interfaces to be replaced.
```suggestion
        get_replace_interfaces_list = self.get_interface_object_for_replaced(new_have, want)
```

Reply:
Thank you for noticing this.

I am committing your suggestion.